Add python3-pyside2.qtquick to python.yaml

knzo25 opened the pull request:
<!-- Thank you for contributing a change to the rosdistro. There are two primary types of submissions.
Please select the appropriate template from below: ROSDEP_RULE_TEMPLATE or DOC_INDEX_TEMPLATE

If you're making a new release with bloom please use bloom to create the pull request automatically.
If you've already run the release bloom has a `--pull-request-only` option you can use.-->


<!-- ROSDEP_RULE_TEMPLATE: Submitter Please review the contributing guidelines: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md -->

Please add the following dependency to the rosdep database.

## Package name:

python3-pyqt5.qtquick

## Package Upstream Source:

None

## Purpose of using this:

Allows creating QML applications.
Although with the current rosdep we can develop QML applications with pyqt due to licensing it would be convenient to also have the PySide2 version available.

Distro packaging links:

## Links to Distribution Packages

<!-- Replace the REQUIRED areas with the URL to the package.  For IF AVAILABLE areas, either put in the URL to the package or state 'not available'.
More info at https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#guidelines-for-rosdep-rules -->

- Debian: https://packages.debian.org/
  - [Link](https://debian.pkgs.org/11/debian-main-amd64/python3-pyside2.qtquick_5.15.2-1_amd64.deb.html)
- Ubuntu: https://packages.ubuntu.com/
   - [Link](https://packages.ubuntu.com/focal/python3-pyside2.qtquick)
- Fedora: https://src.fedoraproject.org/browse/projects/
  - [Link](https://packages.fedoraproject.org/pkgs/python-pyside2/python3-pyside2/)
- Arch: https://www.archlinux.org/packages/
  - [Link](https://archlinux.org/packages/extra/x86_64/pyside2/)
- Gentoo: https://packages.gentoo.org/
  - [Link](https://packages.gentoo.org/packages/dev-python/pyside2)
- macOS: https://formulae.brew.sh/
  - None
- Alpine: https://pkgs.alpinelinux.org/packages
  - None
- NixOS/nixpkgs: https://search.nixos.org/packages
  - None

knzo25 committed:
Adding python3-pyside2.qtquick to rosdep

mabelzhang reviewed (commented):
So, the title of this PR is `python3-pyside2.qtopengl`, but the key and the packages added are `python3-pyside2.qtquick`. Those are different. Which one is it that you want?

The links you provided for upstream sources aren't really to the sources. They're to the documentation.

You didn't provide links for the packages - those "REQUIRED" and "IF AVAILABLE" strings in the template are meant for you to replace with URLs, to help speed up the review process.
For Debian, qtopengl (if that's the one you want) is this for Buster and Bullseye https://packages.debian.org/unstable/python3-pyside2.qtopengl
For Ubuntu, it's this for Focal https://packages.ubuntu.com/focal/python3-pyside2.qtopengl
Etc.

I would suggest that you 1. decide which package it is that you want, 2. look up the links for the other platforms, and 3. add those keys if they exist for those other platforms.

knzo25 commented:
Updated 1, 2, and 3. :bow:

mabelzhang reviewed (commented):
I see that you have updated the title, thanks, but I'd like to point out it still doesn't match the key you're adding, which creates confusion. The title is `python3-pyqt5.qtquick`, which is a key that already exists in line 7618. The key you're adding is `python3-pyside2.qtquick`. Those are different things.

Judging by the description here https://fedora.pkgs.org/34/fedora-x86_64/python3-pyside2-5.15.2-3.fc34.i686.rpm.html and the pattern of the keys around this one (the `.qtopengl` one before it), the general `pyside2` package might satisfy this for arch, fedora, and gentoo. It's a bigger package that includes some `QtQuick*` files in the installation.

Though the key after this one, the `.qtuitools` doesn't have those platforms.

:woman_shrugging: I don't have a strong preference whether you add those, unless someone else has more context about these.

knzo25 committed:
Complemented the entry "python3-pyside2.qtquick" for arch, fedora, and gentoo following the pattern of "python3-pyside2.qtopengl"

knzo25 commented:
Hi, I corrected the PR title, amended the commit title, and added the packages for arch, fedora, and gentoo, following the style of `python3-pyside2.qtopengl`. Thank you so much !

ivanpauno commented:
@knzo25 could you update the PR post with the links to all distribution packages (arch, fedora and gentoo are missing)?
Thanks!

knzo25 commented:
Updated. Thanks ! :slightly_smiling_face:

ivanpauno commented on rosdep/python.yaml:
```diff
@@ -7650,6 +7650,15 @@ python3-pyside2.qtopengl:
+  arch: [pyside2]
+  debian: [python3-pyside2.qtquick]
+  fedora: [python3-pyside2]
+  gentoo: [dev-python/pyside2
```
```suggestion
  gentoo: [dev-python/pyside2]
```

ivanpauno replied:
I think CI is failing because of this, LGTM with it corrected.

knzo25 replied:
Added missing bracket

knzo25 committed:
Added missing bracket
